internal/constellation/helm/charts/edgeless/operators/charts/constellation-operator/crds/nodeversion-crd.yaml

```diff
@@ -48,6 +54,11 @@ spec:
                 description: KubernetesComponentsReference is a reference to the ConfigMap
                   containing the Kubernetes components to use for all nodes.
                 type: string
+              maxNodeBudget:
+                description: MaxNodeBudget is the maximum number of nodes that can
+                  be created simultaneously.
+                format: int32
+                type: integer
```

Comment:
Is this problematic for upgrades, since we never upgrade the installed CRDs through helm?

Reply:
No, I think helm has some upgrade magic build in regarding CRDs. But this time we only add one field that is optional, we are fine. Proof: https://github.com/edgelesssys/constellation/actions/runs/11455280858/job/31871905456.
Or do you have a concrete error in mind, I'm missing?

Reply:
As far as I know, Helm simply doesn't do anything if a CRD already exists: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-1-let-helm-do-it-for-you

So if we change the CRDs, running `helm apply` won't actually apply these changes.
From what I can tell, this effectively makes the `maxNodeBudget` option in the nodeversion crd non-functional.
I'm assuming everything still works fine because we default to `1` if the value is not set in the CR

Reply:
Oh, thanks for the hint. Yes, the behavior is (hopefully) exactly the same. I think we are good for now then (also since the e2e test passed). I didn't want to advertise this feature anyway, but even if we do we should add this constraint. 
In the future(tm), we likely want to do what others do (e.g., istio: https://istio.io/latest/docs/setup/upgrade/helm/#canary-upgrade-recommended).